kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/hspa/HspaModem.java

```diff
@@ -478,25 +479,59 @@ public boolean isGpsEnabled() {
         return false;
     }
 
-    // public abstract boolean isSimCardReady() throws KuraException;
-
     protected CommConnection openSerialPort(String port) throws KuraException {
 
         CommConnection connection = null;
-        if (this.m_connectionFactory != null) {
+        if (this.connectionFactory != null) {
             String uri = new CommURI.Builder(port).withBaudRate(115200).withDataBits(8).withStopBits(1).withParity(0)
                     .withTimeout(2000).build().toString();
 
             try {
-                connection = (CommConnection) this.m_connectionFactory.createConnection(uri, 1, false);
+                connection = (CommConnection) this.connectionFactory.createConnection(uri, 1, false);
             } catch (Exception e) {
-                s_logger.debug("Exception creating connection: " + e);
+                logger.debug("Exception creating connection: " + e);
                 throw new KuraException(KuraErrorCode.INTERNAL_ERROR, e, "Connection Failed");
             }
         }
         return connection;
     }
 
+    @Override
+    public List<ModemPdpContext> getPdpContextInfo() throws KuraException {
+        List<ModemPdpContext> pdpContextInfo = new ArrayList<>();
+        synchronized (atLock) {
+            byte[] reply;
+            CommConnection commAtConnection = openSerialPort(getAtPort());
+            if (!isAtReachable(commAtConnection)) {
+                closeSerialPort(commAtConnection);
+                throw new KuraException(KuraErrorCode.NOT_CONNECTED,
+                        MODEM_NOT_AVAILABLE_FOR_AT_CMDS_MSG + TelitHe910.class.getName());
+            }
+            try {
+                reply = commAtConnection.sendCommand(formGetPdpContextAtCommand().getBytes(), 1000, 100);
+            } catch (IOException e) {
+                closeSerialPort(commAtConnection);
+                throw new KuraException(KuraErrorCode.INTERNAL_ERROR, e);
```

Comment:
I believe we agreed that INTERNAL_ERROR is deprecated. However this code already throws them so we are not making the situation worse. If you can find a better error code, that would be better.